soroban-env-host/src/cost_runner/cost_types/im_map_ops.rs

```diff
@@ -12,20 +12,21 @@ impl CostRunner for ImMapNewRun {
     }
 }
 
-pub struct ImMapImmutEntryRun;
+pub struct ImMapEntryRun;
 #[derive(Clone)]
-pub struct ImMapImmutEntrySample {
+pub struct ImMapEntrySample {
     pub map: HostMap,
     pub keys: Vec<RawVal>,
 }
-impl CostRunner for ImMapImmutEntryRun {
+impl CostRunner for ImMapEntryRun {
     const COST_TYPE: CostType = CostType::MapEntry;
-    type SampleType = ImMapImmutEntrySample;
+    type SampleType = ImMapEntrySample;
 
-    fn run_iter(host: &crate::Host, iter: u64, sample: Self::SampleType) {
-        let _ = sample
-            .map
-            .get(&sample.keys[iter as usize % sample.keys.len()], host)
-            .unwrap();
+    fn run_iter(host: &crate::Host, _iter: u64, sample: Self::SampleType) {
+        for k in &sample.keys {
+            sample.map.get(k, host).unwrap();
```

Comment:
I think this will multiply the iteration count by the key count unnecessarily? We used to just use the iteration value as an index into the key array. Why change it?

Reply:
I thought more iterations would help making the average more accurate but that is probably unnecessary. I've changed it back.